lib/lightning_web/live/project_live/index.ex

```diff
@@ -52,9 +52,97 @@ defmodule LightningWeb.ProjectLive.Index do
   end
 
+  @impl true
+  def handle_event(
+        "delete_now",
+        %{"id" => project_id},
+        socket
+      ) do
+    project = Projects.get_project(project_id)
+
+    if can_delete_project(socket.assigns.current_user, project) do
+      Projects.delete_project(project)
+      |> case do
+        {:ok, _} ->
+          {:noreply,
+           socket
+           |> assign(
+             active_menu_item: :projects,
+             projects: Projects.list_projects(),
+             page_title: "Projects"
+           )
+           |> put_flash(:info, "Project deleted successfully")
+           |> push_patch(to: ~p"/settings/projects")}
+
+        {:error, changeset} ->
+          {:noreply,
+           socket
+           |> assign(changeset: changeset)
+           |> put_flash(:error, "Can't delete project")}
+      end
+    else
+      {:noreply,
+       put_flash(
+         socket,
+         :error,
+         "You are not authorized to perform this action."
+       )
+       |> push_patch(to: ~p"/settings/projects")}
+    end
+  end
+
+  def handle_event(
+        "cancel_delete",
+        %{"id" => project_id},
+        socket
+      ) do
+    project = Projects.get_project(project_id)
+
+    case Projects.cancel_scheduled_deletion(project) do
+      {:ok, _project} ->
+        {:noreply,
+         socket
+         |> put_flash(:info, "Canceled project deletion schedule")
+         |> push_patch(to: ~p"/settings/projects")}
+
+      {:error, %Ecto.Changeset{} = changeset} ->
+        {:noreply, assign(socket, :scheduled_deletion_changeset, changeset)}
+    end
+  end
+
+  def handle_event(
+        "schedule_delete",
+        %{"id" => project_id},
+        socket
+      ) do
+    project = Projects.get_project(project_id)
+
+    case Projects.schedule_project_deletion(project) do
+      {:ok, _project} ->
+        {:noreply,
+         socket
+         |> put_flash(:info, "Project scheduled for deletion")
+         |> push_patch(to: ~p"/settings/projects")}
+
+      {:error, %Ecto.Changeset{} = changeset} ->
+        {:noreply, assign(socket, :scheduled_deletion_changeset, changeset)}
+    end
+  end
+
   # TODO: this results in n+1 queries, we need to precalculate the permissions
   # and have zipped list of projects and the permissions so when we iterate
   # over them in the templace we don't generate n number of queries
+  def can_request_delete_project(current_user, project),
+    do:
+      ProjectUsers
+      |> Permissions.can?(:request_delete_project, current_user, project)
+
   def can_delete_project(current_user, project),
-    do: ProjectUsers |> Permissions.can?(:delete_project, current_user, project)
+    do: Users |> Permissions.can?(:delete_project, current_user, project)
+
+  def request_project_deletion(current_user, project) do
+    is_nil(project.scheduled_deletion) and
+      (can_request_delete_project(current_user, project) or
+         current_user.role == :superuser)
+  end
```

Comment:
Why are we checking for `:superuser` here? All of the permission logic should be inside the policies.
The `:request_delete_project` action/permission should be able to handle the superuser case.

Reply:
Done